defi-pulse/src/data-providers/cryptocompare.ts

```diff
@@ -0,0 +1,48 @@
+import { Requester } from '@chainlink/external-adapter'
+import { Index } from '../adapter'
+import { util } from '@chainlink/ea-bootstrap'
+
+const getPriceData = async (symbols: string) => {
+  const url = 'https://min-api.cryptocompare.com/data/pricemulti'
+  const params = {
+    tsyms: 'USD',
+    fsyms: symbols,
+    api_key: util.getRequiredEnv('API_KEY'),
+  }
+  const config = {
+    url,
+    params,
+  }
+  const response = await Requester.request(config)
+  return response.data
+}
+
+const toAssetPrice = (data: Record<string, any>) => {
+  const price = data['USD']
+  if (!price || price <= 0) {
+    throw new Error('Invalid price')
+  }
+  return price
+}
+
+const getPriceIndex = async (index: Index): Promise<Index> => {
+  const symbols: string[] = []
+  index.forEach(({ asset }) => {
+    symbols.push(asset.toUpperCase())
+  })
```

Comment:
Again, defining data as a `const`! this time, and them mutating that data. This is an antipattern. Always while manipulating data think about transforming data from one form into another, not mutating the source to fit the destination.

```suggestion
  const symbols: string = index.map(({asset})  => (asset)).join().toUpperCase()
```